cdap-proto/src/main/java/io/cdap/cdap/proto/operationrun/OperationError.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.operationrun;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Error representation for Operation Run.
+ */
+public class OperationError {
+
+  private final String message;
+  private final Set<OperationResourceScopedError> details;
```

Comment:
It's more flexible if this is a Collection. Or is this intentionally a Set because we want to enforce only one error per resource?

Reply:
Changing to collection. We don't really need to enforce one error per resource